clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -121,7 +121,8 @@
     /* all the state related to transactions, in particular the producer id, producer epoch, and sequence numbers */
     private final TransactionManager transactionManager;
 
-    public Sender(KafkaClient client,
+    public Sender(LogContext logContext,
```

Comment:
Should Builder pattern be used for the Sender ?
That way the code is more readable when new parameter is added.